pre_processing/structure/get_high_confidence_region_from_AF2.py

```
@@ -0,0 +1,105 @@
+
+    nums = sorted( set( positions ) )
+    # Get start, end positions for each continous confident patch.
+    gaps = [[x, y] for x, y in zip( positions, positions[1:] ) if x+1 < y]
+    edges = iter( positions[:1] + sum( gaps, [] ) + positions[-1:] )
+    return list( zip( edges, edges ) )
+
+
+def merge_confident_regions( confident_patches, pae ):
+    # Merge confident residue patches if PAE <= 5.
+
+    f = open( pae, "r" )
```

Ruff (0.8.2)

Use a context manager for opening files

(SIM115)